src/Symfony/Bridge/PhpUnit/DeprecationErrorHandler.php

```diff
@@ -339,4 +175,216 @@ private static function hasColorSupport()
+                    $v = \dirname(\dirname($r->getFileName()));
+                    if (file_exists($v.'/composer/installed.json')) {
+                        $vendors[] = $v;
+                    }
+                }
+            }
+        }
+        $realPath = realpath($path);
+        if (false === $realPath && '-' !== $path && 'Standard input code' !== $path) {
+            return true;
+        }
+        foreach ($vendors as $vendor) {
+            if (0 === strpos($realPath, $vendor) && false !== strpbrk(substr($realPath, \strlen($vendor), 1), '/'.\DIRECTORY_SEPARATOR)) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * @param string $str
+     * @param bool   $red
+     * @param mixed  $mode
+     *
+     * @return string
+     */
+    private static function colorize($str, $red, $mode)
+    {
+        if (!self::hasColorSupport() || DeprecationErrorHandler::MODE_WEAK === $mode) {
+            return $str;
+        }
+
+        $color = $red ? '41;37' : '43;30';
+
+        return "\x1B[{$color}m{$str}\x1B[0m";
+    }
+
+    /**
+     * @param mixed[] $left
+     * @param mixed[] $right
+     *
+     * @return int
+     */
+    private static function compareDeprecations($left, $right)
+    {
+        return $right['count'] - $left['count'];
+    }
+
+    /**
+     * @param string[] $groups
+     * @param mixed    $mode
+     *
+     * @return int
+     */
+    private static function displayDeprecations($groups, $mode)
+    {
+        foreach ($groups as $group) {
+            if (self::$deprecations[$group.'Count']) {
+                echo "\n", self::colorize(
+                    sprintf('%s deprecation notices (%d)', ucfirst($group), self::$deprecations[$group.'Count']),
+                    'legacy' !== $group && 'remaining vendor' !== $group,
+                    $mode
+                ), "\n";
+
+                uasort(self::$deprecations[$group], [self::class, 'compareDeprecations']);
+
+                foreach (self::$deprecations[$group] as $msg => $notices) {
+                    echo "\n  ", $notices['count'], 'x: ', $msg, "\n";
+
+                    arsort($notices);
+
+                    foreach ($notices as $method => $count) {
+                        if ('count' !== $method) {
+                            echo '    ', $count, 'x in ', preg_replace('/(.*)\\\\(.*?::.*?)$/', '$2 from $1', $method), "\n";
+                        }
+                    }
+                }
+            }
+        }
+        if (!empty($notices)) {
+            echo "\n";
+        }
+    }
+
+    /**
+     * @return string
+     */
+    private static function utilPrefix()
+    {
+        if ('' !== self::$utilPrefix) {
+            return self::$utilPrefix;
+        }
+
+        return self::$utilPrefix = (class_exists('PHPUnit_Util_ErrorHandler') ? 'PHPUnit_Util_' : 'PHPUnit\Util\\');
+    }
+
+    public static function handleError($type, $msg, $file, $line, $context = array())
```

Comment:
Error handlers have to be public and I cannot use `\Closure::fromCallable` here

Reply:
Marked it as internal to discourage its usage.

Reply:
moving this method before colorize() would help review the diff when ignoring whitespaces